File: 2015/5/sol.js
const fs = require('fs');

function ImportFile(fileName)
{
    var file = fs.readFileSync(fileName, 'utf8');
    lines = file.split("\n");

    lines.forEach((line) => {
        line = line.trim();
    });
    return lines;
}

function PartOne(data) {
    let forbidden = ["ab", "cd", "pq", "xy"];
    let vowels = ["a", "e", "i", "o", "u"];
    let sum = 0;

    for (let i = 0; i < data.length; i++) {
        
        if (forbidden.some(fb=>data[i].includes(fb))) continue;

        let double = false;
        for (let j = 0; j < data[i].length -1; j++) {
            if (data[i][j] === data[i][j+1]) {
                double = true;
                break;
            }
        }
        if (!double) continue;
        let count = { a: 0, e: 0, i: 0, o: 0, u: 0}
        for (let j = 0; j < vowels.length; j++) {
            for (let k = 0; k < data[i].length; k++) {
                if (data[i][k] === vowels[j]) {
                    count[vowels[j]]++;
                }
            }
        }
        if (Object.values(count).reduce((a,b)=>a+b) < 3) continue;
        
        sum++;

    }
    return sum;
}


function PartTwo(data)
{
    let sum = 0;
    for (let i = 0; i < data.length; i++) {
        let doubles = false;
        let sandwich = false;
        for (let j = 0; j < data[i].length -1; j++ )
        {
            let i1 = data[i].indexOf(data[i][j] + data[i][j+1]);
            let i2 = data[i].lastIndexOf(data[i][j] + data[i][j+1]);
            if (i1 !== i2 && (i1 +1 !== i2)) {
                doubles = true;
                break;
            }
        }

        if (!doubles) continue;

        for (let j = 0; j < data[i].length -2; j++ )
        {
            if (data[i][j] === data[i][j+2]) {
                sandwich = true;
                break;
            }
        }

        if (!sandwich) continue;
        sum++;
    }
    return sum;
}
let data = ImportFile("text.txt");
console.log(PartOne(data));
console.log(PartTwo(data));

File: 2016/3/sol.py
def ImportFile(fileName):
    return open(fileName, "r").read().split('\n')


def Count(data):
    count = 0
    for i in data:
        i = i.split(' ')
        i = [x for x in i if x != '']
        if (int(i[0]) + int(i[1]) > int(i[2])) and (int(i[0]) + int(i[2]) > int(i[1])) and (int(i[1]) + int(i[2]) > int(i[0])):
            count += 1

    return count


def Count_Vertical(data):
    count = 0
    for i in range(0, len(data), 3):
        r1 = data[i].split(' ')
        r1 = [x for x in r1 if x != '']
        r2 = data[i+1].split(' ')
        r2 = [x for x in r2 if x != '']
        r3 = data[i+2].split(' ')
        r3 = [x for x in r3 if x != '']
        for j in range(len(r1)):
            if (int(r1[j]) + int(r2[j]) > int(r3[j])) and (int(r1[j]) + int(r3[j]) > int(r2[j])) and (int(r2[j]) + int(r3[j]) > int(r1[j])):
                count += 1

    return count


data = ImportFile("text.txt")
print(Count(data))
print(Count_Vertical(data))
